### ISSUE TITLE
An exported `TSImportEqualsDeclaration` should be wrapped in an `ExportNamedDeclaration` instead of having `.isExport = true`

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#4130]
Code from #4129

```ts
namespace Bar {
  export const bar = 123;
}

export namespace Foo {
  export import TheBar = Bar;
}
```

I didn't know that `export import` was even valid at all in TS.
Right now this will emit a `TSImportEqualsDeclaration` with `.isExported = true`.
This is wrong because it means that rules have to have special handling specifically for this case in order to understand `TheBar` is exported.

We should breaking change fix the AST here so that the AST is instead `ExportNamedDeclaration > TSImportEqualsDeclaration` to align with everything else.

**Comment by JoshuaKGoldberg** [comment]
Resolved because #5885 was merged into v6. This'll be released with the v6 version.
